Fixed the ComponentSpec schema

Ark-kun opened the pull request:
There were several syntax/structure errors (`CommandlineArgumentOrArrayType` instead of `StringOrPlaceholder`, Kubernetes schema refs and the top ref).
Also removed couple of obsolete items (source, schemaVersion).

numerology commented:
/lgtm

Do you mind adding some details about the intention of this PR in the description. For example, which PR brought changes that need to be reflected in this schema.

Ark-kun commented:
> Do you mind adding some details about the intention of this PR in the description. For example, which PR brought changes that need to be reflected in this schema.

Added. This change is not in response to some recent PR - mostly fixing very old errors and inconsistencies.

There were several syntax/structure errors (`CommandlineArgumentOrArrayType` instead of `StringOrPlaceholder`, Kubernetes schema refs and the top ref).
Also removed couple of obsolete items (source, schemaVersion).

Ark-kun commented:
/approve
